subprojects/core/src/main/java/org/gradle/internal/service/scopes/VirtualFileSystemServices.java

```diff
@@ -156,9 +159,10 @@ VirtualFileSystem createVirtualFileSystem(
                 FileHasher hasher,
                 ListenerManager listenerManager,
                 Stat stat,
+                FileSystem fileSystem,
                 StringInterner stringInterner
             ) {
-                VirtualFileSystem virtualFileSystem = new DefaultVirtualFileSystem(hasher, stringInterner, stat, DirectoryScanner.getDefaultExcludes());
+                VirtualFileSystem virtualFileSystem = new DefaultVirtualFileSystem(hasher, stringInterner, stat, fileSystem.isCaseSensitive() ? CASE_SENSITIVE : CASE_INSENSITIVE, DirectoryScanner.getDefaultExcludes());
```

Comment:
Nit: please break this up into multiple lines.

Reply:
Done.